### DISCUSSION TITLE
Lack of decent auth options for self-hosted

### DISCUSSION THREAD

**Discussion by rossigee** [hoppscotch/hoppscotch#4294] [General]
Yesterday, I deployed Hoppscotch onto my home lab Kubernetes cluster to play with it. It is now running as expected, and I'm looking forward to trying it out, but I can't log into the admin area to proceed further with configuring it.

None of the existing four auth providers are suitable or appropriate for use in my enviroment (as a home lab hobbyist), and I'm reluctant to spend any time trying to set them up even temporarily for evaluation purposes based on the number of unresolved issues related to self-hosted auth I see in the issue tracker.

Most of my other inhouse services are configured to use Keycloak server in one way or another, and ideally Hoppscotch would too. However, the Hoppscotch README suggests that SSO is only available if you use the Enterprise version :roll_eyes:

I see there has been a proposal for a 'username/password' auth method. I think this is a good idea, as it would make Hoppscotch a potential option for a far wider audience. At least, that would present a usable workaround in my situation while SSO is not (yet?) available.

Long-term, the best solution (imho) would be to simply support OIDC client authentication. But if I want that any time soon I would have to fork and implement it myself based on one of the existing vendor-specific auth methods.

Also, I'm curious, given the existence of the Enterprise edition, and the fact that all auth-related issues in the tracker have since been tagged as 'self-hosted' and otherwise ignored, it begs the question of whether the upstream devs still have any interest in supporting or developing code that would benefit the self-hosted scenarios, or would actively or passively reject it.

For example, if someone went ahead and implemented user/password auth and PR'ed it for the rest of the community, would the devs review and merge such a PR in a timely fashion? 

Or, more controversially, let's say I go ahead and fork Hoppscotch and implement SSO myself to scratch my own itch, and I submitted a PR. How would the devs feel about that? Unfortunately, the possible outcomes I see there are...

1) It would be welcomed.
1a) Devs would review, comment and maybe eventually merge PR.
1b) Devs would comment, pay lip service for appearance purposes, but constantly delay or never actually merge.

2) It would be considered a commercial 'conflict of interest' in light of the Enterprise offering, and...
2a) Strategically ignored.
2b) Attacked and take-down requests issued etc.

Knowing the devs position on this would help me (and others) to make decisions on the suitability of Hoppscotch as a part of our development toolset.

**Comment by TheCodingWanderer** [comment]
I don't think the developers are really interrested in accepting PRs that adds SSO capabilities to the free version since this is one of the most important features in the enterprise version. This strategy is understandable from the perspective of Hoppscotch as a commercial organization. 
So if you can fork hoppscotch and add the SSO feature and maybe keepup merging changes from hoppscotch, this would be a nice thing. I don't have the needed skills to do so.

**Comment by krystalcode** [comment]
In my case I don't even need authentication, I just want to run a local instance that is not accessible to anybody other than myself in my LAN. I had issues with creating an account using Email auth provider. All that I want is to store data in the database instead of the the browser local data - where they can be lost.

A root user (super admin) could be created by default with username/password set via environment variables, that's a common approach. It would also serve recover if you lose access as a developer.
